### PR TITLE
Dev datamigration microsoft.data migration 2020 09 01 preview

### PR DESCRIPTION
**Pull request by vasundhraManhas**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ashutoshsuman99** [comment]
Adding @rchlkm

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2020-06-14/examples/KustoAttachedDatabaseConfigurationsCreateOrUpdate.json [hunk 1]
```diff
@@ -1,61 +1,61 @@
-{
-  "parameters": {
-    "subscriptionId": "12345678-1234-1234-1234-123456789098",
```
Comment (hitenjava) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2020-06-14/examples/KustoAttachedDatabaseConfigurationsCreateOrUpdate.json:3]: Looks like some git merge issue or formatting change?
Why is this file showing up - This should not be changed.

Comment (vasundhraManhas) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2020-06-14/examples/KustoAttachedDatabaseConfigurationsCreateOrUpdate.json:3]: There was some styling/formatting error (Prettier check) and the suggestion  was to use npm run prettier-fix
I think it fixed the formatting issues in all the files

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/examples/SqlMiCreateOrUpdateDatabaseMigrationMIN.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
+  "parameters": {
+    "scope": "subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/testrg/providers/Microsoft.Sql/managedInstances/managedInstance1",
+    "targetDbName": "db1",
+    "api-version": "2020-09-01-preview",
+    "parameters": {
+      "location": "northeurope",
+      "properties": {
+        "sourceDatabaseName": "aaa",
+        "scope": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/testrg/providers/Microsoft.Sql/managedInstances/instance",
```
Comment (j5lim) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/examples/SqlMiCreateOrUpdateDatabaseMigrationMIN.json:10]: >scope [](http://example.com/codeflow?start=9&length=5)

Is this different from the scope parameter?

This should be returned in the response. (all properties in the put should be returned except secrets)

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/sqlmigration.json [hunk 3]
```diff
@@ -0,0 +1,1718 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Get Database Migration resource.": {
+            "$ref": "./examples/SqlMiGetDatabaseMigration.json"
+          }
+        }
+      },
+      "put": {
```
Comment (j5lim) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/sqlmigration.json:69]: >"put": [](http://example.com/codeflow?start=6&length=6)

No PATCH? If a partial update is supported, it must be via PATCH.

[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc009-use-put-for-update-of-entire-resource-ie-replace-use-patch-only-for-partial-update-of-resource](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc009-use-put-for-update-of-entire-resource-ie-replace-use-patch-only-for-partial-update-of-resource)

Comment (ashutoshsuman99) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/sqlmigration.json:69]: Currently we don't support partial update on Database Migration resource. @hitenjava can you please add further details on this.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/examples/SqlMiCreateOrUpdateDatabaseMigrationMIN.json (hunk 2 above)
Comment (j5lim) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/examples/SqlMiCreateOrUpdateDatabaseMigrationMIN.json:10]: Why scope needs to be specified both in the URL (line 3) and the body (line 10)? Are they different?

If the scope in the body is needed in the request body, the response also should return it.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/sqlmigration.json (hunk 3 above)
Comment (hitenjava) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2020-09-01-preview/sqlmigration.json:69]: Correct - databaseMigrations resource is a proxy resource and we don't support partial update on databaseMigrations resource.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2021-10-30-preview/sqlmigration.json [hunk 4]
```diff
@@ -0,0 +1,1718 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "List database migrations attached to the service.": {
+            "$ref": "./examples/ListMigrationsByMigrationService.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataMigration/sqlMigrationServices/{sqlMigrationServiceName}/getMonitoringData": {
```
[the remark this answers is not in the document]
Comment (vasundhraManhas) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2021-10-30-preview/sqlmigration.json]: @pilor GET getMonitoringData call has been changed to POST "listMonitoringData" call

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2021-10-30-preview/sqlmigration.json [hunk 5]
```diff
@@ -0,0 +1,1718 @@
...
+    "DatabaseMigration": {
+      "description": "Database Migration Resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "location": {
```
[the remark this answers is not in the document]
Comment (ashutoshsuman99) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2021-10-30-preview/sqlmigration.json]: Hi @pilor, we have removed the location property for DatabaseMigration.

**Comment by vasundhraManhas** [comment]
azp run

**Comment by ArcturusZhang** [comment]
The cross version breaking changes are false alarms

**Comment by raych1** [comment]
@ArcturusZhang , `model validation` has errors in examples. Please fix them.
https://github.com/Azure/azure-rest-api-specs/pull/16032/checks?check_run_id=4439591756

**Comment by vasundhraManhas** [comment]
> @ArcturusZhang , `model validation` has errors in examples. Please fix them. https://github.com/Azure/azure-rest-api-specs/pull/16032/checks?check_run_id=4439591756

You are referring to an earlier commit. These errors have been fixed.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2021-10-30-preview/definitions/Common.json [hunk 6]
```diff
@@ -291,6 +291,22 @@
           "type": "string",
           "description": "Data source in the format Protocol:MachineName\\SQLServerInstanceName,PortNumber"
         },
+        "serverName": {
+          "type": "string",
+          "description": "name of the server"
+        },
+        "serverBrandVersion": {
```
Comment (ash1625) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2021-10-30-preview/definitions/Common.json:298]: Didn't see serverBrandVersion flagged in s360 report for SqlConnectionInfo object

Comment (vasundhraManhas) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2021-10-30-preview/definitions/Common.json]: > Didn't see serverBrandVersion flagged in s360 report for SqlConnectionInfo object

@ash1625 thanks, removed now

**Review by ArcturusZhang — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**